apps/aecore/lib/aecore/peers/sync.ex

```diff
@@ -208,31 +208,39 @@ defmodule Aecore.Peers.Sync do
   end
 
   # Gets all unknown blocks, starting from the given one
-  defp check_peer_block(peer_uri, block_hash, state) do
-    case Chain.has_block?(block_hash) do
+  defp check_peer_blocks(peer_uri, from_block_hash, state) do
+    case Chain.has_block?(from_block_hash) do
       false ->
-        case(HttpClient.get_block({peer_uri, block_hash})) do
-          {:ok, deserialized_block} ->
-            try do
-              BlockValidation.single_validate_block(deserialized_block)
-              peer_block_hash =
-                BlockValidation.block_header_hash(deserialized_block.header)
-
-              if(block_hash == peer_block_hash) do
-                check_peer_block(peer_uri, deserialized_block.header.prev_hash,
-                  Map.put(state, peer_block_hash, deserialized_block))
-              else
-                state
-              end
-            catch
-              {:error, _} ->
-                state
+        case HttpClient.get_raw_blocks({peer_uri, from_block_hash, Chain.top_block_hash()}) do
+          {:ok, blocks} ->
+            if !Enum.empty?(blocks) do
+              state = Enum.reduce(blocks, state, fn(block, state_acc) ->
+                try do
+                  BlockValidation.single_validate_block(block)
```

Comment:
this is a complex function, can you make it less deep, by extracting another function?